versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/condition/MongoDBConditionAliasVisitor.java

```diff
@@ -0,0 +1,113 @@
+  public ConditionExpression visit(ConditionExpression conditionExpression, AliasCollector collector) {
+    return ImmutableConditionExpression.builder()
+      .functions(conditionExpression.getFunctions().stream()
+        .map(f -> f.acceptExpressionFunction(this, collector)).collect(ImmutableList.toImmutableList()))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param expressionFunction The object to be aliased.
+   * @param arguments constituent arguments of the object to be aliased.
+   * @param name the name of the object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return the aliased ExpressionFunction.
+   */
+  @Override
+  public ExpressionFunction visit(ExpressionFunction expressionFunction, List<Value> arguments,
+                                  ExpressionFunction.FunctionName name, AliasCollector collector) {
+    return new ExpressionFunction(name, arguments.stream().map(v -> getArgumentValue(v, collector))
+      .collect(ImmutableList.toImmutableList()));
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param expressionpath The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return The aliased ExpressionPath.
+   */
+  @Override
+  public ExpressionPath visit(ExpressionPath expressionpath, AliasCollector collector) {
+    return ImmutableExpressionPath.builder()
+      .root((ExpressionPath.NameSegment) expressionpath.getRoot().accept(this, collector))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param nameSegment The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return The aliased ExpressionPath.
+   */
+  @Override
+  public ExpressionPath.NameSegment visit(ExpressionPath.NameSegment nameSegment, AliasCollector collector) {
+    return ImmutableNameSegment.builder()
+      .name(collector.escape(nameSegment.getName()))
+      .child(nameSegment.getChild().map(p -> p.alias(collector)))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param value The object to be aliased.
+   * @param entity ValueOfEntity value
+   * @param collector the class that does the aliasing.
+   * @return The aliased Value
+   */
+  @Override
+  public Value visit(Value value, Entity entity, AliasCollector collector) {
+    return ExpressionPath.builder(collector.alias(entity)).build();
+  }
+
+  /**
+   * Returns an aliased sub class equivalent to the type of Value.
+   * @param value the value to alias.
+   * @param collector the class that does the aliasing.
+   * @return The aliased ExpressionPath.
+   */
+  Value getArgumentValue(Value value, AliasCollector collector) {
+    switch (value.getType()) {
+      case PATH:
+        return value.getPath().acceptExpressionPath(this, collector);
+      case VALUE:
+        return value.acceptValue(this, collector);
```

Comment:
why acceptValue() and not accept()?

Reply:
There are two "accept"s in ExpressionPath for Value and ExpressionPath. I had to rename one as the method signatures would have been identical. I'm hoping exposing required internals of those classes required will mean I only require one accept, to be called `accept`.

Reply:
I kept the acceptValue but acceptExpressionPath -> accept()